packages/gateway/src/features/registration/root-resolvers.ts

```diff
@@ -636,20 +639,79 @@ export const resolvers: GQLResolver = {
 
       return taskEntry.resource.id
     },
-    async confirmRegistration(_, { id }, { headers: authHeader }) {
+    async confirmRegistration(_, { id, details }, { headers: authHeader }) {     
+      const token = authHeader.Authorization.replace("Bearer ", "").trim();
+      const recordSpecificToken = await tokenExchangeHandler(
+        token,
+        authHeader,
+        id
+      )
+
+      authHeader.Authorization = `Bearer ${recordSpecificToken.access_token}`
+
       if (!inScope(authHeader, ['record.confirm-registration'])) {
-        throw new Error(
-          'User does not have a "record.confirm-registration" scope'
-        )
+        throw new Error('User does not have a Confirm Registration scope');
+      }
+
+      if (!hasRecordAccess(authHeader, id)) {
+        throw new Error('User does not have access to the record');
+      }
+
+      try {
+        const taskEntry = await confirmRegistration(
+          id,
+          authHeader,
+          {
+            error: details.error,
+            registrationNumber: details.registrationNumber,
+            childIdentifiers: details.identifiers,
+            compositionId: id
+          }
+        );
+
+        return taskEntry.resource.id;
+      } catch (error) {
+        throw new Error(`Failed to confirm registration: ${error.message}`);
+      }
+    },
+    async rejectRegistration(
+      _,
+      { id, details },
+      { headers: authHeader }
+    ) {
+
+      const token = authHeader.Authorization.replace("Bearer ", "").trim();
+      const recordSpecificToken = await tokenExchangeHandler(
+        token,
+        authHeader,
+        id
+      )
+
+      authHeader.Authorization = `Bearer ${recordSpecificToken.access_token}`
```

Comment:
Same here as in confirmRegistration

Reply:
Does this mean i have to remove this part? if i do so when i try to test it I am getting the user does not have the scope error message.

Reply:
https://github.com/opencrvs/opencrvs-core/pull/7882 we finally got the e2e environment set up properly so I got this merged 👍 you can pull it from develop

Reply:
ok will take the pull, thank you for the update👍